### PR TITLE
Adds REFRESH_DB param and updates refresh-data script to add conditional

### PR DESCRIPTION
**Pull request by thephillipsequation**
cfgov-refresh's docker-stack.yml currently runs the refresh-data.sh every time. This is unnecessary considering the postgres service does not redeploy when just app code changes.

This causes a couple issues:

Data modifications are lost between deploys
It makes the deploys slow. It is an expensive operation to refresh the database, pulling down the big DB archive and applying it to the running container.

## Additions

- REFRESH_DB parameter in the Jenkinsfile that defaults to false
- Conditional logic added to the refresh-data.sh script

## Removals

-

## Changes

-

## Testing

Run an adhoc deploy with refresh db = true
Check for django wagtail table in postgres (wagtailcore_page suggested)
Run an adhoc deploy with refresh db = false
Check for django wagtail table in postgres (wagtailcore_page suggested)
## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [ ] Chrome on desktop
- [ ] Firefox
- [ ] Safari on macOS
- [ ] Edge
- [ ] Internet Explorer 9, 10, and 11
- [ ] Safari on iOS
- [ ] Chrome on Android

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: docker-stack.yml [hunk 1]
```diff
@@ -65,7 +66,18 @@ services:
...
+            if django-admin inspectdb wagtailcore_page | grep \"db_table = \'wagtailcore_page\'\" && $REFRESH_DB == false;
+              then 
+                echo 'Skipping DB refresh as db already exists'
+                httpd -d cfgov/apache -D FOREGROUND
+            elif $REFRESH_DB == true;
+              then
+                ./refresh-data.sh && httpd -d cfgov/apache -D FOREGROUND
+            else
+                ./refresh-data.sh && httpd -d cfgov/apache -D FOREGROUND
+            fi "
```
Comment (higs4281) [inline docker-stack.yml:80]: If both elif and else result in the same action, couldn't you just use `else` ?

```suggestion
            else
                ./refresh-data.sh && httpd -d cfgov/apache -D FOREGROUND
            fi "
```

Also, I wonder if using psql could be a simpler test of database existence:

```
psql -l | grep cfgov
```

Comment (thephillipsequation) [reply docker-stack.yml]: modified the logic a bit to make the conditional more explicit

File: refresh-data.sh [hunk 2]
```diff
@@ -8,6 +8,17 @@
...
 
+if django-admin inspectdb wagtailcore_page | grep "db_table = 'wagtailcore_page'" && $REFRESH_DB == false;
+    then 
+        echo 'Skipping DB refresh as db already exists'
+        exit 1
+    elif $REFRESH_DB == true;
+        then
+            continue
+    else
+        continue
```
Comment (willbarton) [inline refresh-data.sh:19]: Both macOS's `bash` and `bash` in the `python` container complain about this test the use of `continue`:

```
./refresh-data.sh: line 15: ==: command not found
./refresh-data.sh: line 19: continue: only meaningful in a `for', `while', or `until' loop
```

This also seems superfluous, as if the original test for the database falls through, the script will continue regardless.

```suggestion
```

**Comment by willbarton** [comment]
s> This is unnecessary considering the postgres service does not redeploy when just app code changes.

I have one other thought — it might be worth having the `./refresh-data.sh` script run migrations even if set not to refresh the database. I think there are pros and cons to doing that automatically though. I'd be curious what @chosak thinks.

**Comment by chosak** [comment]
@willbarton it seems like your comment is out of date since the changes to refresh-data.sh were removed in the latest commits here? Regardless, I vote that a script named refresh-data.sh should, indeed, refresh data. 😄

File: refresh-data.sh [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 set -e
 
+
```
Comment (chosak) [inline refresh-data.sh:11]:
```suggestion
```

Comment (thephillipsequation) [reply refresh-data.sh]: removed the line :)

File: docker-compose.prod.yml [hunk 4]
```diff
@@ -24,4 +24,4 @@ services:
             - ./cfgov/apache/conf/:/src/cfgov-refresh/cfgov/apache/conf/
             - ./cfgov/apache/conf.d/:/src/cfgov-refresh/cfgov/apache/conf.d/
             - ./cfgov/apache/conf.modules.d/:/src/cfgov-refresh/cfgov/apache/conf.modules.d/
-        
+
```
Comment (Scotchester) [inline docker-compose.prod.yml:27]: Mind reverting the changes to this file for a cleaner history?

Comment (thephillipsequation) [reply docker-compose.prod.yml:27]: sure I can do that

File: Dockerfile [hunk 5]
```diff
@@ -71,16 +71,17 @@ ENV DJANGO_SETTINGS_MODULE cfgov.settings.production
 ENV DJANGO_STATIC_ROOT ${STATIC_PATH}
 ENV ALLOWED_HOSTS '["*"]'
 
+
+# See .dockerignore for details on which files are included
+COPY --chown=apache:apache . .
+
```
Comment (Scotchester) [inline Dockerfile:77]: This change seems unrelated to the purpose of this PR? Doesn't seem like it would be harmful, though.

File: docker-stack.yml [hunk 6]
```diff
@@ -65,7 +66,20 @@ services:
...
         - com.docker.lb.network=${LB_NETWORK}
-    command: /bin/bash -c './refresh-data.sh && httpd -d cfgov/apache -D FOREGROUND'
+    command: >
+                bash -c "
+                    
+                psql $${DATABASE_URL} -c 'SELECT COUNT(*) FROM auth_user'
+                
+                django_tables_exist=$$?
+                
+                echo $$django_tables_exist
```
[the remark this answers is not in the document]
Comment (thephillipsequation) [reply docker-stack.yml]: removed

File: Dockerfile (hunk 5 above)
Comment (thephillipsequation) [reply Dockerfile]: sorry this leaked in from a bad commit from a separate branch, I've reverted it!
